docs/DeveloperGuide.md

```diff
@@ -302,28 +302,169 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 (For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case: UC01 - List all contacts**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User issues the list command.
+2.  The system retrieves and displays the list of all contacts to the user.
+
+    Use case ends.
+
+
+
+**Use case: UC02 - Add a contact**
+
+**MSS**
+
+1.  User issues the add-contact command with the corresponding details.
+2.  The system adds the contact and displays a success message.
+3.  The system shows the new contact in the address book.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. The system detects a name input error (duplicated or trailing whitespace).
+    * 1a1. The system displays an error message.
+  
+        The use case ends.
+  
+  
+* 1b. The system detects a phone number input error (invalid format): 
+    * 1b1. The system displays an error message stating the correct format.
+  
+        The use case ends.
+  
+  
+* 1c. The system detects an address input error (too long):
+    * 1c1. The system displays an error message stating the maximum length.
+  
+        The use case ends.
+  
+  
+* 1d. The system detects an email input error (invalid format):
+    * 1d1. The system displays an error message stating the correct format.
+  
+        The use case ends.
+  
+  
+* 1e. The system detects a duplicate phone number error:
+    * 1e1. The system displays an error message mentioning the existence of a duplicate phone number.
+    
+        The use case ends.
+        
+
+* 1f. The system detects an invalid tag input:
+    * 1f1. The system displays an error message stating the tag is invalid.
+    
+        The use case ends.
+
+
+
+**Use case: UC03 - Add Phone Number to Contact**
+
+**Guarantees:**
+* No duplicate phone numbers will exist across different contacts.
+
+
+**MSS**
+
+1. User <ins>lists all contacts (UC01)</ins>.
+2. User issues the add-phone command with the corresponding details.
+3. The system adds the phone number to the contact and displays a success message.
+4. The system displays the updated contact information in the address book.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. The list is empty.
+
+  The use case ends.
+
+
+* 2a. The system detects an invalid contact index:
+    * 2a1. The system displays an error message stating the contact index is invalid.
+
+        The use case resumes at step 2.
```

Comment:
Similarly, maybe without "the"?

Reply:
If this extension is branched from Step 2, perhaps we could phrase it as "Use case repeats from step 2" or something similar?

Reply:
Sure, I will change it to

> Use case resumes at step 2.

I'm not too sure about the word "repeat" as the example describes it as "resumes" for a similar extension

> Use case: Delete a person
> 
> MSS
> 
> User requests to list persons
> AddressBook shows a list of persons
> User requests to delete a specific person in the list
> AddressBook deletes the person
> 
> Use case ends.
> 
> Extensions
> 
> 2a. The list is empty.
> 
> Use case ends.
> 
> 3a. The given index is invalid.
> 
> 3a1. AddressBook shows an error message.
> 
> Use case resumes at step 2.

Reply:
Ooh I did not see that haha Could it be because it branches off at step 3 then goes (back) to step 2 right? Whereas this one branches off at step 2 and repeats step 2? Maybe "resumes" is a safer option after all 👍